### DISCUSSION TITLE
MilvusClient refresh_load()

### DISCUSSION THREAD

**Discussion by HoangSang1510** [milvus-io/milvus#31921] [Q&A and General discussion] [answered]
Hello, I used collection.flush() of ORM/Collection. Now i using MilvusClient, will I use refresh_load() instead?

**Comment by yhmo** [comment, marked as answer]
No need to call collection.flush() manually, milvus can automatically trigger flush action by itself.
If you call collection frequently, there will be lots of tiny segments generated and increase the cost to compact, eventually slow down the performance of milvus.

MilvusClient doesn't have a flush interface. The refresh_lolad() is to force the datacoord to refresh the segments list to update the data snapshot, this is an internal action, you can treat it as a normal load().

**Reply by HoangSang1510** [reply]
Thanks for your answer, i have code below, my 'count_1' will be 0 without using refresh_load()

```client = MilvusClient(uri="http://localhost:19530")
collection_name = 'my_collection'
data=[{"id": 1, "vector": embedding_vec}]
res = client.upsert(
    collection_name=collection_name,
    data=data,
    partition_name="part_1"
)
count_1 = client.query(
    collection_name=collection_name,
    partition_names=["part_1"],
    filter="",
    output_fields=["count(*)"]
)
print('count_1:  ', count_1)```

**Reply by yhmo** [reply]
Data visibility is controlled by consistency level: https://milvus.io/docs/consistency.md#Consistency-levels

Each collection has a consistency level. The default level is "Bounded", which means allows data inconsistency during a certain period of time.
You can set a consistency level when you call create_collection() method:
```
client.create_collection(
  collection_name=xxx,
  dimension=xxx,
  vector_field_name=xxx,
  ......
  consistency_level="Strong",
)
```

"Strong" level ensure that the recent insert data can be seachable. But with this level the search latency will increase 200ms.
